storage/txcache/concurrentMap.go

```diff
@@ -143,3 +143,34 @@ func (m *ConcurrentMap) Clear() {
 	m.chunks = make([]*concurrentMapChunk, m.nChunks)
 	m.mutex.Unlock()
 }
+
+// Keys returns all keys as []string
+func (m *ConcurrentMap) Keys() []string {
```

Comment:
Getting all the keys can be pretty time consuming especially when the txs pool is full.
This I think we currently need because of the cleanup process, which was designed for the previous pool type.
I think we can redesign the cleanup for the new cache type so that we improve the processing time. 

One idea is that when cleaning executed tx hashes for a block, we can remove all the txs with lower nonce from the accounts-txs cache, for the respective account as well.

Can you add a TODO somewhere for this so that we don't forget? maybe where the cleanup is called.